src/OrchardCore/OrchardCore.DisplayManagement/Descriptors/ShapeTemplateStrategy/ShapeTemplateOptionsSetup.cs

```diff
@@ -26,6 +29,10 @@ public void Configure(ShapeTemplateOptions options)
             {
                 options.FileProviders.Add(_hostingEnvironment.ContentRootFileProvider);
             }
+
+            // To let the application behaving as a module, its views are requested under the virtual
+            // "Areas" folder, but there are still served from the file system by this custom provider.
+            options.FileProviders.Insert(0, new ApplicationViewFileProvider(_applicationContext));
```

Comment:
Looks like this provider uses the file system. If this is true shouldn't it only work in development?

Reply:
Yes, because it is related to the application's module which has no embedded files as other modules. At the application level razor views can be precompiled (when publishing) but not embedded, application level liquid view can only be in the file system.

It was part of the embedded file provider which is used both for views and shapes. But i extracted this part in a dedicated file provider that we used for views but i forgot to do the same for shapes.

That said, application level **precompiled** views work for `@page` and mvc views, but not for a shape, to be discovered the file need to be in the file system, for other modules there are still discovered because there are embedded.